src/flask/blueprints.py

```diff
@@ -453,6 +453,15 @@ def extend(bp_dict, parent_dict):
         for blueprint, bp_options in self._blueprints:
             bp_options = bp_options.copy()
             bp_url_prefix = bp_options.get("url_prefix")
+            bp_subdomain = bp_options.get("subdomain")
+
+            if bp_subdomain is None:
+                bp_subdomain = blueprint.subdomain
+
+            if state.subdomain is not None and bp_subdomain is None:
```

Comment:
I think we should also consider what to do if `state.subdomain is not None and bp_subdomain is not None`. I think we should combine as `state.bp` (much like with the prefix). Would you expect it to work this way as well?

Reply:
Yes good point.

I thought that if both parent and child have subdomain set, it would be weird to combine. It makes sense for paths but subdomains in the form foo.bar.baz are rare.

For example if parent is bar and child is foo, we would get foo.bar.

But I can get this by supplying foo.bar in the child blueprint subdomain. So I went with the child overrides parent subdomain way that way the user can use a combined subdomain as needed.

Not sure this is a good idea though because some use cases might need them to be combined by default.

Another possible solution would be to introduce an extra parameter called something like "combine_subdomains". This will also allow different child blueprint to have different subdomain strategies. For example, foo can choose to be combined but baz can choose not to.

What do you think?

Reply:
Given it might confuse it is probably safest to raise a ValueError/RuntimeError then if both are set. (This would be raised during initialization and so isn't likely to cause any issues for existing apps)

Reply:
It could be an issue if the user wants to combine the subdomain anyway.

For example, parent blueprint's subdomain is *bar* then user wants child blueprint X to be available at "foo.bar" so he sets the subdomain. Then another blueprint Y needs to be at "baz.bar" so he sets the subdomain like that. But parent needs to be available at *bar* subdomain no matter what.

Raising ValueError would break this use case.

Reply:
I've had quite a bit of time to think about this, and to talk to others about their expectations. I think we should combine the subdomains (much like we do for the paths). I think this is the clearest expectation for a nested blueprint i.e. that the routes defined within it nest both by path and subdomain. Could you update this PR to do so and then I'll merge.